### ISSUE TITLE
Problem upgrading 0.1.7 to 0.2.0

### ISSUE THREAD

**Issue by sillevl** [Ideetron/Lorank#8]
When upgrading from 0.1.7 to 0.2.0 I get the following error

```
root@lorank8:~# ./Lorank/lorank8v1/upgrade 0.2.0
~/Lorank ~
Already on 'master'
Already up-to-date.
~
./Lorank/lorank8v1/upgrade: line 34: pushd: Lorank/Ngrok: No such file or directory
```

note: I did a manually checkout of master just before calling the update script, but I think it should not mather (or am I wrong?)

Looking into the update script, I see the update code/procedure of Ngrok is changed since 0.1.7. I don't know if It could cause this issue.

**Comment by devlaam** [comment]
Thanks for your comments. I will look into this asap.

**Comment by sillevl** [comment]
I just checked on a gateway preinstalled with 0.2.0 and I see an `Loriot` and `Ngrok` directory inside `/root/Lorank`. These directories are not available on the 0.1.7 gateway. Seems like this might be the reason why the `pushd` command is failing.

**Comment by devlaam** [comment]
Well the reason is that a command like mkdir -p Lorank/Ngrok is failing. Unfortunately, there are a couple of other 'issues'. The point is this upgrade script should work starting independent of the version number. Hard to test with limited resources.

**Comment by sillevl** [comment]
I just searched the repository, but cannot find any command `mkdir -p Lorank/Ngrok`. It should indeed solve the problem if it works correctly.

I created the directories manually ( `mkdir Lorank/Ngrok` and `mkdir Lorank/Loriot`). Then ran the update script again and it installed and compiled everything without errors.
Only when running the web interface (after reboot) it seems to be the old 0.1.7 version instead of the 0.2.0 interface

**Comment by sillevl** [comment]
I think the upgrade might be successful afterall. I removed the old directory and everything still works.
I was confused about the `index.html` file that provides the web interface. This file was not copied to `/var/lib/cloud9` after the update (still have the old version). I have no logs of the output of the update script. Looking at the code, it seems that the `init_bbb` script might not been executed or did not execute correctly.

**Comment by sillevl** [comment]
Just did an upgrade to 0.2.1 (from 0.2.0) and everything worked well. The only thing that I notice is that the `index.html` page is not updated. The old version remains in `/var/lib/cloud9/`. I will create a new issue to report this. (https://github.com/Ideetron/Lorank/issues/9)

**Comment by francisdb** [comment]
I had a similar issue upgrading from `0.1.6` to `0.2.2``

```
...
gcc obj/util_tx_test.o obj/base64.o -o util_tx_test
make[1]: Leaving directory `/root/lorank8-0.2.2/packet_forwarder/util_tx_test'
~
~/lorank8v1 ~
WARNING: New global_conf.json, manually copy your entries from the old version, if needed.
~
mv: cannot stat `Lorank/Loriot/loriot_pkt_fwd': No such file or directory
```

same solution as @sillevl 

fixed by running

```
mkdir ~/Lorank/Ngrok
mkdir Lorank/Loriot
./Lorank/lorank8v1/upgrade 0.2.2
```

**Comment by devlaam** [comment]
Thanks for the fix francisdb, since we are now on 0.2.3 and this is solved in this and its prior version, this can be closed.
